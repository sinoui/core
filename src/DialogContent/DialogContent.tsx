import DialogTitle from '@sinoui/core/DialogTitle';
import Typography from '@sinoui/core/Typography';
import getColorFromTheme from '@sinoui/core/utils/getColorFromTheme';
import classNames from 'classnames';
import styled from 'styled-components';

const DialogContent = styled(Typography).attrs(({ className }) => ({
  type: 'subtitle1',
  className: classNames('sinoui-dialog-content', className),
  color: 'textSecondary',
}))`
  padding: 24px 24px 24px 24px;
  color: ${(props) => getColorFromTheme(props.theme, props.color)};
  overflow-y: auto;
  flex: 1 1 auto;
  ${DialogTitle} ~ & {
    padding-top: 0;
  }
`;

export default DialogContent;
