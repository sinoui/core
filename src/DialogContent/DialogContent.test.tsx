import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { ThemeProvider } from 'styled-components';
import { defaultTheme } from '@sinoui/theme';
import DialogContent from '@sinoui/core/DialogContent';

describe('DialogContent组件 快照测试', () => {
  it('基本使用', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={defaultTheme}>
          <DialogContent>
            Let Google help apps determine location. This means sending
            anonymous location data toGoogle, even when no apps are running.
          </DialogContent>
        </ThemeProvider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
