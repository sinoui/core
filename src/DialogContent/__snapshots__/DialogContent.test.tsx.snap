// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DialogContent组件 快照测试 基本使用 1`] = `
<h6
  className="Typography__TypographyWrapper-sc-12agh80-0 juBaHC sinoui-typography sinoui-typography--subtitle1 DialogContent-rvt14v-0 coPUFm sinoui-dialog-content children forwardedComponent theme"
  color="textSecondary"
  type="subtitle1"
>
  Let Google help apps determine location. This means sending anonymous location data toGoogle, even when no apps are running.
</h6>
`;
